client/state/country-states/reducer.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	COUNTRY_STATES_RECEIVE,
+	COUNTRY_STATES_REQUEST,
+	COUNTRY_STATES_REQUEST_FAILURE,
+} from 'state/action-types';
+import { itemSchema } from './schema';
+import { createReducer } from 'state/utils';
+
+// Stores the complete list of states, indexed by locale key
+export const items = createReducer( {}, {
+	[ COUNTRY_STATES_RECEIVE ]: ( state, action ) => ( { ...state, [ action.countryCode ]: action.countryStates } ),
+}, itemSchema );
+
+// Tracks states list fetching state
+export const isFetching = createReducer( {}, {
+	[ COUNTRY_STATES_REQUEST ]: ( state, { countryCode } ) => ( { ...state, [ countryCode ]: true } ),
+	[ COUNTRY_STATES_RECEIVE ]: ( state, { countryCode } ) => ( { ...state, [ countryCode ]: false } ),
```

Comment:
Part of the idea with a separate `receiveCountryStates` action creator is that it allows for states to be received from any source, not specifically tied to the request from `requestCountryStates`. This is why in other subtrees we'll have a separate action type. In this case that would be `COUNTRY_STATES_REQUEST_SUCCESS`.